appcues/src/main/java/com/appcues/statemachine/Transitions.kt

```diff
@@ -169,14 +170,16 @@ private fun transitionsToBeginningStep(
 ) = if (experience.areStepsFromDifferentGroup(currentStepIndex, nextStepIndex)) {
     // given that steps are from different container, we now get step container index to present
     experience.groupLookup[nextStepIndex]?.let { stepContainerIndex ->
+        val stepGroup = experience.stepContainers[stepContainerIndex]
+        val actions = stepGroup.actions[stepGroup.id]?.filter { it.on == NAVIGATE }?.map { it.experienceAction } ?: emptyList()
```

Comment:
to make it more readable we could make this an extension function that has a clear name like stepGroup.getNavigationActions(stepContainerIndex)